nab/labeler.py

```diff
@@ -194,52 +208,62 @@ def write(self, destPath):
 
 
   def combine(self):
-    """Combine the raw user labels, and set the anomaly windows."""
-    self.getUserLabels()
-    self.combineRawLabels()
+    """Combine raw and known labels in anomaly windows."""
+    self.getRawLabels()
+    self.combineLabels()
     self.applyWindows()
     self.checkWindows()
-    self.addKnownLabels()
 
 
-  def getUserLabels(self):
-    """Collect the raw user labels from default labels directory."""
+  def getRawLabels(self):
+    """Collect the raw user labels from specified directory."""
     labelPaths = absoluteFilePaths(self.labelDir)
+    self.userLabels = []
+    self.knownLabels = []
+    for path in labelPaths:
+      if "known" in path:
+        self.knownLabels.append(CorpusLabel(path, self.corpus))
+      else:
+        self.userLabels.append(CorpusLabel(path, self.corpus))
 
-    self.userLabels = [CorpusLabel(path, self.corpus) for path in labelPaths]
-
-    if len(self.userLabels) == 0:
+    self.nLabelers = len(self.userLabels)
+    if self.nLabelers == 0:
       raise ValueError("No users labels found")
 
-    self.nlabelers = len(self.userLabels)
-
     
-  def combineRawLabels(self):
+  def combineLabels(self):
     """
-    Combines raw user labels to create set of true anomaly labels; checks 
-    explicitly for common start times in the raw windows.
-    A buffer is used to merge labels that identify the same anomaly. The buffer
+    Combines raw user labels to create set of true anomaly labels.
+    A buffer is used to bucket labels that identify the same anomaly. The buffer
     is half the estimated window size of an anomaly -- approximates an average
     of two anomalies per dataset, and no window can have >1 anomaly.
-    After merging, a label becomes a true anomaly if it was labeled by a
-    proportion of the users greater than the defined threshold.
+    After bucketing, a label becomes a true anomaly if it was labeled by a
+    proportion of the users greater than the defined threshold. Then the bucket
+    is merged into one timestamp -- the ground truth label.
+    The set of known anomaly labels are added as well.
     
     If verbosity>0, the dictionary passedLabels -- the raw labels that did not
     pass the threshold qualification -- is printed to the console.
     """
-    combinedLabels = {}
-    labelIndices = {}
-    passedLabels = {}
+    def setTruthLabels(dataSet, trueAnomalies):
```

Comment:
Because it's only called in combineLabels() I thought it would be tidier, similar style to found() in getWindows(). No problem to move it outside the method though.